### PR TITLE
Add Behat tests to validate metadata endpoints

### PR DESCRIPTION
**Pull request by VadimSchmitz**
The following work has been done:
- Add  Behat tests to validate metadata endpoints
- Re-add second_factor_only.metadata_factory service

Task 1 of [181710432](https://www.pivotaltracker.com/story/show/181710432)

The build currently fails due to a security vulnerability

### CODE REVIEW

File: src/Surfnet/StepupGateway/SecondFactorOnlyBundle/Resources/config/services.yml [hunk 1]
```diff
@@ -33,11 +33,19 @@ services:
...
+            - "@router"
+            - "@surfnet_saml.signing_service"
+            - "@second_factor_only.configuration.metadata"
+
     second_factor_only.response_rendering:
         class: Surfnet\StepupGateway\GatewayBundle\Service\ResponseRenderingService
         arguments:
             - "@gateway.proxy.response_builder"
-            - "@templating"
+            - "@twig"
```
Comment (MKodde) [inline src/Surfnet/StepupGateway/SecondFactorOnlyBundle/Resources/config/services.yml:48]: :tennis: :lady_beetle:

File: tests/features/metadata.feature [hunk 2]
```diff
@@ -8,3 +7,27 @@ Feature: As a SP or IdP
     Then the response should match xpath '//md:EntityDescriptor[@entityID="https://gateway.stepup.example.com/authentication/metadata"]'
     And the response should match xpath '//md:SingleSignOnService[@Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"]'
     And the response should match xpath '//md:SingleSignOnService[@Location="https://gateway.stepup.example.com/authentication/single-sign-on"]'
+
+  Scenario: View the SecondFactor metadata of the Gateway
```
Comment (MKodde) [inline tests/features/metadata.feature:11]: Nit: ~SecondFactor~ second factor only

File: tests/features/metadata.feature [hunk 3]
```diff
@@ -8,3 +7,27 @@ Feature: As a SP or IdP
...
+    Given I am on "/second-factor-only/metadata"
+    Then the response should match xpath '//md:EntityDescriptor[@entityID="https://gateway.stepup.example.com/second-factor-only/metadata"]'
+    And the response should match xpath '//md:SingleSignOnService[@Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"]'
+    And the response should match xpath '//md:SingleSignOnService[@Location="https://gateway.stepup.example.com/second-factor-only/single-sign-on"]'
+
+  Scenario: View the GSSP Azuremfa metadata of the Gateway
```
Comment (MKodde) [inline tests/features/metadata.feature:17]: Nit: `Azuremfa` Azure MFA

File: tests/features/metadata.feature [hunk 4]
```diff
@@ -8,3 +7,27 @@ Feature: As a SP or IdP
...
+    And the response should match xpath '//md:SingleSignOnService[@Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"]'
+    And the response should match xpath '//md:SingleSignOnService[@Location="https://gateway.stepup.example.com/gssp/azuremfa/single-sign-on"]'
+
+  Scenario: View the GSSP Tiqr metadata of the Gateway
+    Given I am on "/gssp/tiqr/metadata"
+    Then the response should match xpath '//md:EntityDescriptor[@entityID="https://gateway.stepup.example.com/gssp/tiqr/metadata"]'
+    And the response should match xpath '//md:SingleSignOnService[@Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"]'
+    And the response should match xpath '//md:SingleSignOnService[@Location="https://gateway.stepup.example.com/gssp/tiqr/single-sign-on"]'
+
+  Scenario: View the GSSP webauthn metadata of the Gateway
```
Comment (MKodde) [inline tests/features/metadata.feature:29]: Nit: ~webauthn~ WebAuthn

**Review by MKodde — APPROVED** [review]
Very nice work! Thanks for reverting the mess-up we made in the previous PR. 

See some cosmetic feedback below.

File: tests/features/metadata.feature (hunk 2 above)
Comment (VadimSchmitz) [reply tests/features/metadata.feature:11]: changed

File: tests/features/metadata.feature (hunk 3 above)
Comment (VadimSchmitz) [reply tests/features/metadata.feature:17]: changed

File: tests/features/metadata.feature (hunk 4 above)
Comment (VadimSchmitz) [reply tests/features/metadata.feature:29]: changed

File: tests/features/metadata.feature [hunk 5]
```diff
@@ -8,13 +8,13 @@ Feature: As a SP or IdP
     And the response should match xpath '//md:SingleSignOnService[@Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"]'
     And the response should match xpath '//md:SingleSignOnService[@Location="https://gateway.stepup.example.com/authentication/single-sign-on"]'
 
-  Scenario: View the SecondFactor metadata of the Gateway
+  Scenario: View the SecondFactor of the Gateway
```
Comment (MKodde) [inline tests/features/metadata.feature:11]: Oops: I think this should read: `View the second factor only metadata of the Gateway`

**Pull request merged**